### DISCUSSION TITLE
Is it possible to promote a preview deployment to production using the CLI?

### DISCUSSION THREAD

**Discussion by denik1981** [vercel/vercel#6423] [General]
The question don't need any further explanation. I just want to promote a preview deployment from using the command line (not the API client) to production. In most cases it will be the last deployment. Is this possible? How can I do it? 

I'm not referring to deploying from my local with the flag "--prod" but from the already available preview deployments that are in vercel. I know that this can be accomplish by pushing to the master branch or whatever branch you have linked to your prod deployments. It happens that sometimes I cannot push to master because that can mess up everything else with my other branches.

**Comment by barelyhuman** [comment]
As for your exact question, you can go through https://vercel.com/docs/cli#commands and if you already did go through it, you know that there's no direct way other than maybe the api route. You can check if there's already a similar feature request in the https://github.com/vercel/vercel/issues section, if not , i guess you can raise one and the team might respond accordingly.

Though I think the problem you state is more of a code management thing though? 

Possible solutions that I think I'd follow would be 
- changing how you handle branches and un-tested code 
- using `git tag` to make checkpoints in the repository so you can move back and forth and run the `vercel --prod` for each accordingly. 


cause technically vercel does the same thing, when you do it on the web portal, it takes in reference the commit hash and clone up till there then run the build step again if needed. 
This doesn't happen if the cache is still valid and just moves it up to production. 

Hope that was helpful :)
